### PR TITLE
feat(eap): rough protobuf timeseries API

### PR DESCRIPTION
**Pull request by volokluev**

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -11,6 +11,7 @@ Flask==2.2.5
 google-cloud-storage==2.18.0
 googleapis-common-protos==1.63.2
 google-api-core==2.19.1
+proto-plus==1.24.0
```
Comment (colin-sentry) [inline requirements.txt:14]: this is already in this file

File: snuba/protobufs/base_messages.proto [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  LTE = 4;
+  GTE = 5;
+  LIKE = 6;
+  NOT_LIKE = 7;
+  IN = 8;
+  NOT_IN = 9;
+}
+
+
+message PentityFilter{
```
Comment (evanh) [inline snuba/protobufs/base_messages.proto:28]: At a certain point this will just be a Protobuf version of the objects we have in the Snuba SDK, right?

Comment (volokluev) [reply snuba/protobufs/base_messages.proto:28]: More restrictive. For example we won't allow arbitrary functions here like we do in the snuba SDK

Comment (volokluev) [reply snuba/protobufs/base_messages.proto:28]: I'm actually going to remove these filters and use the ones Colin proposed

https://github.com/getsentry/snuba/blob/master/snuba/protobufs/Filters.proto

File: snuba/datasets/configuration/events_analytics_platform/entities/eap_spans.yaml [hunk 3]
```diff
@@ -54,7 +57,13 @@ storages:
 storage_selector:
   selector: DefaultQueryStorageSelector
 
-query_processors: []
+query_processors:
```
Comment (evanh) [inline snuba/datasets/configuration/events_analytics_platform/entities/eap_spans.yaml:60]: Can we get rid of this? I don't like this processor and would prefer it go away. It does a lot more than provide a `time` alias that this entity doesn't need.

File: snuba/web/rpc/timeseries.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+TimeSeriesResponse = Any
+
+
+def _agg_type_to_ch_function(agg_type: AggregationType) -> str:
+    lookup = {
+        AggregationType.AVG: "avg",
+        AggregationType.MIN: "min",
+        AggregationType.MAX: "max",
+        AggregationType.P50: "quantiles(0.5)",
```
Comment (evanh) [inline snuba/web/rpc/timeseries.py:27]: Can you change this to `quantile` instead? Otherwise you'll get an array in the results.

**Review by evanh — COMMENTED** [review]
I notice that we are not trying to solve how functions can be used in this API, which is fine to start. However how to handle functions has been a difficult problem in all our APIs, and it's one that this API will need to solve as well.

I think figuring that out earlier will be time well spent, because functions are almost certainly going to be something we need to support. For example, your filters won't support `has(tags.key, '...') = 1`, or `fromUTCTimestamp(timestamp_ms) > '...'`.

**Comment by volokluev** [comment]
This PR is no longer necessary

File: snuba/web/rpc/timeseries.py (hunk 4 above)
Comment (volokluev) [reply snuba/web/rpc/timeseries.py:27]: Thank you!

File: snuba/datasets/configuration/events_analytics_platform/entities/eap_spans.yaml (hunk 3 above)
Comment (volokluev) [reply snuba/datasets/configuration/events_analytics_platform/entities/eap_spans.yaml:60]: The TimeSeriesProcessor has been pretty helpful in making these queries easy to write, will on the performance team asked for it as well. What is the concern with it?

**Pull request closed without merging**